Issue/32 missions

fenruga opened the pull request:
@dcversus приде
Как же у меня горело, когда я мержил с девелопом, это жесть

Commit:
issue #32 link checker

Commit:
Merge branch 'develop' of https://github.com/hyperloot/bot.hyperloot.net into issue/32-missions

Commit:
refactoring after master pull

fenruga commented:
![20190305_19 44 05](https://user-images.githubusercontent.com/5465079/53821569-178ca300-3f7f-11e9-9003-a4f9da1f794b.png)

dcversus commented on src/modules/commands/missionAdd.js:
```diff
@@ -44,7 +82,15 @@ const missionAdd = async function (req, ctx) {
+        const { key, values } = JSON.parse(description.replace(/'/g, '"'));
+
+        values.user = assignee;
+        modifyDescriptionValues(values, checker, missionId, assigneeId || assignee);
```
DONT MUTATE!

dcversus commented on src/modules/commands/missionAdd.js:
```diff
@@ -13,7 +14,44 @@ function getDiscordIdFromMention(mention) {
 }
 
+// todo: move it somewhere | prbbly to checker. Unification
+function modifySettings(originalSettings, checker, missionId) {
```
keep meta for all missions?

dcversus commented on src/modules/commands/missionAdd.js:
```diff
@@ -13,7 +14,44 @@ function getDiscordIdFromMention(mention) {
+}
+
+// todo: move it somewhere | prbbly to checker. Unification
+function modifyDescriptionValues(values, checker, missionId, assignee) {
```
its part of checker _INIT_ section... or something like this...

probably, i think, all checkers its linear modules with submodule "checker"... like command.filter module =)

dcversus commented on src/modules/commands/missionAdd.js:
```diff
@@ -44,7 +82,15 @@ const missionAdd = async function (req, ctx) {
+        modifyDescriptionValues(values, checker, missionId, assigneeId || assignee);
+        description = i18n(key, values);
+    } catch (e) {
+        description = originalDescription;
```
feee

fenruga replied:
I know =\

dcversus commented on src/modules/commands/missionAdd.js:
```diff
@@ -55,15 +101,19 @@ const missionAdd = async function (req, ctx) {
         extend(query, { requirements: JSON.parse(requirements.replace(/'/g, '"')) });
     }
 
+    modifyMission(query);
```
mutations(

dcversus commented on src/modules/commands/missionAdd.js:
```diff
@@ -55,15 +101,19 @@ const missionAdd = async function (req, ctx) {
-        extend(query, { checkerSettings: JSON.parse(checkerSettings.replace(/'/g, '"')) });
+        const parsedCheckerSettings = JSON.parse(checkerSettings.replace(/'/g, '"'));
+
+        modifySettings(parsedCheckerSettings, checker, missionId);
```
=(

dcversus commented on src/modules/missions/checkers/linkChecker.js:
```diff
@@ -0,0 +1 @@
+module.exports = ({ missionId }, options) => options.missionId === missionId;
```
... should has some db call specific code? or something that need for link checking

dcversus commented on src/modules/missions/missionChecker.js:
```diff
@@ -99,16 +101,16 @@ module.exports = async function missionChecker(req, ctx) {
             }
 
-            send({
+            req.output = {
```
DONT USE OUTPUT!
`ctx.send`

fenruga replied:
ctx.send f*cks me up =\

dcversus commented on src/modules/missions/missionChecker.js:
```diff
@@ -99,16 +101,16 @@ module.exports = async function missionChecker(req, ctx) {
 
-        if (actualChecker(ctx, mission.checkerSettings) && await checkAndUpdateRequirements(mission, req, ctx)) {
-            if (!mission.requirements.cooldown) {
+        if (actualChecker(req, mission.checkerSettings) && await checkAndUpdateRequirements(mission, req, ctx)) {
```
very dirty

fenruga committed:
Merge branch 'develop' of https://github.com/uz0/u.dcvs.ru into issue/32-missions

fenruga committed:
* endMission
* missions all

Commit:
emptyChecker refactoring

fenruga committed:
manual checker added
empty checker refactoring
link checker refactoring
url added to config
missionCheck command added
i18n for missions/checkers

Commit:
style fix

fenruga committed:
refactored

Commit:
Merge branch 'develop' of https://github.com/hyperloot/bot.hyperloot.net into issue/32-missions

Commit:
i18n fixes
missionAdd help msg
embeds

dcversus commented on src/index.js:
```diff
@@ -48,6 +51,8 @@ instance.use([
     missionAdd,
+    missionCheck,
+    missionEnd,
     missions,
```
seems like all missions details need place inside one mission module

dcversus commented on src/adapters/discord.js:
```diff
@@ -93,6 +99,8 @@ discordAdapter.__INIT__ = function (ctx) {
         });
     };
 
+    discordAdapter.handler = handler;
```
what it? bad solution(

fenruga replied:
undeleted change for prev iteration =\

dcversus commented on src/config.js:
```diff
@@ -3,11 +3,13 @@ module.exports = {
     port: process.env.PORT || process.env.VCAP_APP_PORT || 3000,
     mongoURI: process.env.MONGODB_URI,
-    admin: process.env.ADMINISTRATOR_NICKNAME || 'dcversus',
+    adminDiscordId: process.env.ADMIN_DISCORD_ID || '258540630681190402',
```
bad

fenruga replied:
?

dcversus commented on src/config.js:
```diff
@@ -3,11 +3,13 @@ module.exports = {
     discord: {
         color: '0xf9690e',
         authToken: process.env.DISCORD_AUTH_TOKEN,
+        broadcastChannelName: process.env.BROADCAST_CHANNEL_NAME,
```
its seems like we can get merge conflicts

fenruga replied:
with?
